shorebird_code_push/example/lib/main.dart

```diff
@@ -225,3 +224,43 @@ class _LoadingIndicator extends StatelessWidget {
     );
   }
 }
+
+sealed class AsyncValue<T> {
```

Comment:
This is really neat., but I worry it might make the example harder to follow. I'd either document the absolute crap out of it or remove it in favor of something more simple.